rust/wallet_core_rs/src/ffi/solana/address.rs

```diff
@@ -37,7 +38,47 @@ pub unsafe extern "C" fn tw_solana_address_default_token_address(
     );
 
     let token_address = try_or_else!(
-        StakeProgram::get_associated_token_address(main_address, token_mint_address),
+        StakeProgram::get_associated_token_address(
+            main_address,
+            *TOKEN_PROGRAM_ID_ADDRESS,
+            token_mint_address
+        ),
+        std::ptr::null_mut
+    );
+
+    TWString::from(token_address.to_string()).into_ptr()
+}
+
+/// Derive token 2022 address for token
+///
+/// \param address Non-null pointer to a Solana Address
+/// \param token_mint_address Non-null pointer to a token mint address as a string
+/// \return Null pointer if the token 2022 address for a token is not found, valid pointer otherwise
+#[no_mangle]
+pub unsafe extern "C" fn tw_solana_address_token_2022_address(
```

Comment:
Please consider adding a helper function with the common implementation to avoid repeating yourself.
For example,
```rust
unsafe fn tw_solana_address_token_address_impl(
  address: *const TWString,
  token_mint_address: *const TWString,
  token_address: SolanaAddress,
) -> *mut TWString {
      let main_address = try_or_else!(TWString::from_ptr_as_ref(address), std::ptr::null_mut);
      // TODO
}

#[no_mangle]
pub unsafe extern "C" fn tw_solana_address_default_token_address(
    address: *const TWString,
    token_mint_address: *const TWString,
) -> *mut TWString {
  tw_solana_address_token_address_impl(address, token_mint_address, *TOKEN_PROGRAM_ID_ADDRESS)
}

// Same for `tw_solana_address_token_2022_address`
```

Reply:
Thanks for pointing it out. I have already made the adjustment.